### PR TITLE
Allow to read openstack cloud provider config from a secret

### PR DESCRIPTION
**Pull request by Fedosin**
This PR fixes openstack cloudprovider to read secrets from informer, so as it is consistent with other cloudproviders that need to
read kubernetes objects.

/kind bug

```release-note
Fix openstack cloudprovider to read secrets from informer, so as it is consistent with other cloudproviders that need to
read kubernetes objects
```

### CODE REVIEW

**Comment by sttts** [comment]
Sgtm as a fix-up of an existing carry. Please sync with next rebase team how to handle the upstream PR rename.

**Comment by gnufied** [comment]
afaict - this PR attempts fixing two things (mainly):

1. It removes use of kubeconfig for reading secrets and uses informer. Openstack cloudprovider already supported reading cloud-config from secrets, so this change is not a new feature I think.
2. The second fix is - it uses metadata service to fetch instance related information in kubelet. Does this mean - we don't need cloud-provider configuration in kubelet anymore? Can we mark all functions within cloudprovider that is called by kubelet?

File: plugin/pkg/admission/storage/persistentvolume/label/admission.go [hunk 1]
```diff
@@ -175,7 +193,7 @@ func (l *persistentVolumeLabel) findVolumeLabels(volume *api.PersistentVolume) (
 	topologyLabelGA := true
 	domain, domainOK := existingLabels[v1.LabelTopologyZone]
 	region, regionOK := existingLabels[v1.LabelTopologyRegion]
-	// If they don't have GA labels we should check for failuredomain beta labels
+	// If they dont have GA labels we should check for failuredomain beta labels
```
Comment (mdbooth) [inline plugin/pkg/admission/storage/persistentvolume/label/admission.go:196]: Comment damage.

File: plugin/pkg/admission/storage/persistentvolume/label/admission.go [hunk 2]
```diff
@@ -85,6 +90,19 @@ func newPersistentVolumeLabel() *persistentVolumeLabel {
 func (l *persistentVolumeLabel) SetCloudConfig(cloudConfig []byte) {
 	l.cloudConfig = cloudConfig
 }
```
Comment (mdbooth) [inline plugin/pkg/admission/storage/persistentvolume/label/admission.go:92]: Missing blank line.

**Review by mdbooth — COMMENTED** [review]
It doesn't look like we're watching the cloud secret, so presumably if we update it nothing will notice. Are we ok with that?

File: staging/src/k8s.io/legacy-cloud-providers/openstack/openstack_instances.go [hunk 3]
```diff
@@ -48,6 +48,19 @@ const (
 func (os *OpenStack) Instances() (cloudprovider.Instances, bool) {
 	klog.V(4).Info("openstack.Instances() called")
 
+	// For Kubelet we don't have to initialize cloud provider as it
+	// can read all required data from metadata service. Absence of
+	// secretLister means that we use kubelet.
+	if os.secretLister == nil {
+		return &Instances{
```
Comment (gnufied) [inline staging/src/k8s.io/legacy-cloud-providers/openstack/openstack_instances.go:55]: while testing this change - what I found is, kubelet still needs cloud-config even if it is invalid one. Can we make `cloud-config` an optional parameter for cloud-provider initialization so as kubelet does not crashes if no cloud-config is provided?

Comment (Fedosin) [reply staging/src/k8s.io/legacy-cloud-providers/openstack/openstack_instances.go:55]: Done! now `cloud-provider` is an optional parameter.

**Pull request closed without merging**